src/main/java/org/generationcp/middleware/api/location/LocationServiceImpl.java

```diff
@@ -67,44 +67,18 @@ public long countFilteredLocations(final LocationSearchRequest locationSearchReq
 	@Override
 	public void deleteLocation(final Integer locationId) {
 		final Location location = this.daoFactory.getLocationDAO().getById(locationId);
-		List<Integer> blockIds = null;
-		if (location.getLtype() == LocdesType.FIELD.getId()) {
-			// Get the Blocks
-			blockIds = daoFactory.getLocDesDao().getLocdes(null, Arrays.asList(location.getLocid().toString()))
-				.stream().map(Locdes::getLocationId).collect(Collectors.toList());
-
-			// Delete Block
-			this.daoFactory.getLocDesDao().deleteByLocationIds(blockIds);
-			this.daoFactory.getLocationDAO().deleteByLocationIds(blockIds);
-
-			// Delete Field
-			final List<Integer> fieldIds = daoFactory.getLocDesDao().getLocdes(Arrays.asList(location.getLocid()), null)
-				.stream().map(Locdes::getLocationId).collect(Collectors.toList());
-			this.daoFactory.getLocDesDao().deleteByLocationIds(fieldIds);
-		} else if (location.getLtype() == LocdesType.BLOCK.getId()) {
-			blockIds = daoFactory.getLocDesDao().getLocdes(Arrays.asList(location.getLocid()), null)
-				.stream().map(Locdes::getLocationId).collect(Collectors.toList());
-
-			// Delete Block
-			this.daoFactory.getLocDesDao().deleteByLocationIds(blockIds);
-
+		if (location.getLtype().equals(LocdesType.FIELD.getId())) {
```

Comment:
switch de order of equals comparison to prevent NPE @cuenyad

Reply:
done.